ui/src/app/applications/components/application-parameters/application-parameters.tsx

```diff
@@ -646,21 +646,21 @@ function gatherCoreSourceDetails(i: number, attributes: EditablePanelItem[], sou
         } else {
             attributes.push({
                 title: 'TARGET REVISION',
-                view: <Revision repoUrl={source.repoURL} revision={source.targetRevision || 'HEAD'} />,
-                edit: (formApi: FormApi) => <RevisionFormField helpIconTop={'0'} hideLabel={true} formApi={formApi} repoURL={source.repoURL} fieldValue={revisionField} />
+                view: <Revision repoUrl={source?.repoURL} revision={source?.targetRevision || 'HEAD'} />,
```

Comment:
This is an example of a strange place to resort to nil coalescing. If `source` is nil, the revision isn't `HEAD`, it's "unknown." Seems like we should handle the failure state explicitly rather than falling back to a seemingly-happy code path.